tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,179 @@
+__all__ = ["FastTextModel"]
+
+
+import gc
+import logging
+import psutil
+import tempfile
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+from autogluon.core.utils.loaders import load_pkl
+from autogluon.core.utils.savers import save_pkl
+
+from ...features.feature_metadata import FeatureMetadata
+from ...features.feature_metadata import S_TEXT
+
+from autogluon.core.constants import BINARY, MULTICLASS
+from ..abstract.abstract_model import AbstractModel
+from .hyperparameters.parameters import get_param_baseline
+
+logger = logging.getLogger(__name__)
+
+np.random.seed(0)
+
+def try_import_fasttext():
+    try:
+        import fasttext
+
+        _ = fasttext.__file__
+    except Exception:
+        raise ImportError('Import fasttext failed. Please run "pip install fasttext"')
+
+
+class FastTextModel(AbstractModel):
+    model_bin_file_name = "fasttext.ftz"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._model_bin_available = False
+
+    def _set_default_params(self):
+        default_params = get_param_baseline()
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                required_special_types=[S_TEXT],
```

Comment:
Testing around by also including the categorical integer encoded columns here, I notice that I can sometimes get quite significant accuracy improvements with the categorical columns. I'd rather keep things as they are now, but it might be good to add a `# TODO: Investigate allowing categorical features as well`